worker/upgradedatabase/upgradedatabase.go

```diff
@@ -77,6 +88,15 @@ func (w *Worker) upgradeDone() bool {
 		return true
 	}
 
+	// If we are already on the current version, there is nothing to do.
```

Comment:
Put the `loop` function before `upgradeDone` so that reading down has higher levels of abstraction first.